sys/net/network_layer/ng_ipv6/ng_ipv6.c

```diff
@@ -432,7 +442,13 @@ static void _send(ng_pktsnip_t *pkt, bool prep_hdr)
     else {
         ng_ipv6_addr_t *next_hop = NULL;
 
-        next_hop = &hdr->dst;   /* TODO: next hop determination */
+#ifdef MODULE_NG_IPV6_EXT_RH
+        next_hop = ng_ipv6_ext_rh_next_hop(hdr);
```

Comment:
>How is it specified as a destination and by whom?

The destination address and the (current address in SRH)++ are swapped constantly until the packet reaches its final destination (last address in the SRH address list).

>Who caches the SRH beforehand so I can use it here? And wouldn't some kind of SRH cache add code overhead too?

I was actually just thinking of a plain pointer to the routing header `ng_ipv6_ext_rh_t`, which was processed on the receiving side, when I referred to a cache. This pointer could be used on the sending side - no need to iterate ovre all extension headers to check for a routing header.